submitit/core/utils.py

```diff
@@ -117,6 +117,7 @@ def __init__(self, function: Callable[..., Any], *args: Any, **kwargs: Any) -> N
         self._result: Any = None
         self._done = False
         self.timeout_countdown: int = 0  # controlled in submission and execution
+        self.sys_path = list(sys.path)
```

Comment:
you should probably hide it, and use a tuple to make it unmutable.
I am not sure I like it. I am actually pretty sure I don't :D but if that fixes it, then why not (modulo the other comment...)